src/hooks/useGatewayDelegates.ts

```diff
@@ -0,0 +1,39 @@
+import { AoGatewayDelegateWithAddress } from '@ar.io/sdk/web';
+import { useGlobalState } from '@src/store';
+import { useQuery } from '@tanstack/react-query';
+
+const useGatewayDelegateStakes = (address?: string) => {
+  const arIOReadSDK = useGlobalState((state) => state.arIOReadSDK);
+
+  const res = useQuery({
+    queryKey: ['gatewayDelegates', address, arIOReadSDK],
+    queryFn: async () => {
+      if (!address) {
+        throw new Error('Address is not set');
+      }
+
+      let cursor: string | undefined;
+
+      let results: Array<AoGatewayDelegateWithAddress> = [];
+
+      do {
+        const pageResult = await arIOReadSDK.getGatewayDelegates({
```

Comment:
Just a note on this pattern - we can improve this pagination by sorting to a specific type and caching each page respectively

Reply:
we would then cache those pages with no stale time and then search for new pages where we return all the previous pages and query for the latest, if latest retrieved then we add that to a new cursor cache and it gets updated accordingly.